wallet-new/src/Cardano/Wallet/API/Request/Pagination.hs

```diff
@@ -0,0 +1,102 @@
+{- | Support for resource pagination.
+-}
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE RankNTypes                 #-}
+module Cardano.Wallet.API.Request.Pagination (
+      Page(..)
+    , PerPage(..)
+    , maxPerPageEntries
+    , defaultPerPageEntries
+    , PaginationMetadata(..)
+    , PaginationParams(..)
+    ) where
+
+import           Universum
+
+import           Data.Aeson.TH
+import           Data.Default
+import qualified Serokell.Aeson.Options as Serokell
+import           Test.QuickCheck (Arbitrary (..), choose, getPositive)
+import           Web.HttpApiData
+
+-- | A `Page` is used in paginated endpoints to request access to a particular
+-- subset of a collection.
+newtype Page = Page Int
```

Comment:
Maybe we could consider `Word64` as it is unsigned (and we will have twice as much pages).

If we will need more pages then `Integer` seems the way to go

Reply:
@akegalj If we consider a very generous `10` entries per pages, we can represent up to `214748364` pages with an `Int`. Seems quite a lot for the foreseeable future. What do you think? 😉